### PR TITLE
ENG-11594: Guard against nothing on deque and PBD

### PR DESCRIPTION
**Pull request by akhanzode**

### CODE REVIEW

File: src/frontend/org/voltdb/export/ExportDataSource.java [hunk 1]
```diff
@@ -315,12 +315,13 @@ public synchronized void updateAckMailboxes(final Pair<Mailbox, ImmutableList<Lo
 
     private void releaseExportBytes(long releaseOffset) throws IOException {
         // if released offset is in an already-released past, just return success
-        if (!m_committedBuffers.isEmpty() && releaseOffset < m_committedBuffers.peek().uso()) {
+        if (!m_committedBuffers.isEmpty() && (m_committedBuffers.peek() != null)
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: What happens when "m_committedBuffers.peek() == null"?

Comment (xinjiacs) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: Can you do early return on " If (m_committedBuffers.isEmpty())"? Then you do not need to change them in the next while loop?

Comment (akhanzode) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: we just wait for export push to offer() and then when release comes after we cleanup. This is a very edge case and not easy to reproduce.

Comment (xinjiacs) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: All right, I am familiar with code base. But this PR seems to be better on NULL check case. ""m_committedBuffers.peek() == null"".

Comment (akhanzode) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: Didnt quite follow what you mean.

**Pull request merged**